test/distributed/org/apache/cassandra/distributed/mock/nodetool/NetstatsTest.java

```diff
@@ -0,0 +1,607 @@
+
+        public void add(String[] result)
+        {
+            netstatOutputs.add(result);
+        }
+
+        public void assertSuccessful()
+        {
+            for (final String[] result : netstatOutputs)
+            {
+                InstanceAwareNodetool.assertSuccessful(result);
+            }
+        }
+    }
+
+    private static class NetstatsCallable implements Callable<NetstatResults>
+    {
+        private final IInvokableInstance node;
+        private final InstanceAwareNodetool nodetool;
+
+        public NetstatsCallable(final InstanceAwareNodetool nodetool,
+                                final IInvokableInstance node)
+        {
+            this.nodetool = nodetool;
+            this.node = node;
+        }
+
+        public NetstatResults call() throws Exception
+        {
+            final NetstatResults results = new NetstatResults();
+
+            boolean sawAnyStreamingOutput = false;
+
+            while (true)
+            {
+                String[] result = nodetool.execute(node, "netstats");
+
+                if (!sawAnyStreamingOutput)
+                {
+                    if (result[0].contains("Receiving") || result[0].contains("Sending"))
+                    {
+                        sawAnyStreamingOutput = true;
+                    }
+                }
+
+                if (sawAnyStreamingOutput && (!result[0].contains("Receiving") && !result[0].contains("Sending")))
+                {
+                    break;
+                }
+
+                results.add(result);
+
+                Thread.currentThread().sleep(1000);
+            }
+
+            return results;
+        }
+    }
+
+    private void changeReplicationFactor()
+    {
+        try (com.datastax.driver.core.Cluster c = com.datastax.driver.core.Cluster.builder().addContactPoint("127.0.0.1").build();
+             Session s = c.connect())
+        {
+            s.execute("ALTER KEYSPACE netstats_test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 2};");
+        }
+    }
+
+    private void createTable(Cluster cluster)
+    {
+        // replication factor is 1
+        cluster.schemaChange("CREATE KEYSPACE netstats_test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};");
+        cluster.schemaChange("CREATE TABLE netstats_test.test_table (id uuid primary key);");
+    }
+
+    private void populateData()
+    {
+        try (com.datastax.driver.core.Cluster c = com.datastax.driver.core.Cluster.builder().addContactPoint("127.0.0.1").build();
+             Session s = c.connect("netstats_test"))
+        {
+
+            for (int i = 0; i < 100000; i++)
+            {
+                s.execute("INSERT INTO test_table (id) VALUES (" + UUID.randomUUID() + ")");
+            }
+        }
+    }
+
+    private void disableCompaction(IInvokableInstance invokableInstance)
+    {
+        invokableInstance.runOnInstance(() -> {
```

Comment:
can do

```
invokableInstance.nodetoolResult("disableautocompaction", "netstats_test").asserts().success()
```

Reply:
@dcapwell interesting enough, if I remember correctly, this has not worked. I dont know why but sstables still got compacted, or maybe it was instance.disableCompaction() or something like that ... it just didnt work